src/raw/aws/fargate_metadata.go

```diff
@@ -157,3 +158,17 @@ func MetadataV3BaseURL() (*url.URL, error) {
 	}
 	return parsedURL, nil
 }
+
+// MetadataV4BaseURL returns the v4 metadata endpoint configured via the ECS_CONTAINER_METADATA_URI environment
+// variable.
+func MetadataV4BaseURL() (*url.URL, error) {
+	baseURL, found := os.LookupEnv(containerMetadataEnvVarV4)
+	if !found {
+		return nil, fmt.Errorf("could not find env var with Metadata V4 API URL: %s", containerMetadataEnvVarV4)
+	}
+	parsedURL, err := url.Parse(baseURL)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse Metadata V4 API URL (%s): %s", baseURL, err)
```

Comment:
for instance if v4 fails because of this error , we are going to use v3 and not see any error